README.md

```diff
@@ -2064,6 +2108,48 @@ We have implemented a small SPA app, located at `./test/app/` which is used inte
 
 Because this test app is set up to dynamically change configuration and leak internal information, users should not use this test app as the basis for their own applications. Instead, use the example usage outlined elsewhere in this README.
 
+## Migrating from previous versions
+
+The [CHANGELOG](CHANGELOG.md) contains details for all changes and links to the original PR.
+
+### From 3.x to 4.x
+
+* Now using named exports. You should change code like
+
+```javascript
+import OktaAuth from '@okta/okta-auth-js'
+```
+
+to
+
+```javascript
+import { OktaAuth } from '@okta/okta-auth-js'
+```
+
+If using CommonJS, change
+
+```javascript
+const OktaAuth = require('@okta/okta-auth-js');
+```
+
+to
+
+```javascript
+const OktaAuth = require('@okta/okta-auth-js').OktaAuth;
+```
+
+* Typescript definitions are now included. If you were providing your own definitions for `OktaAuth` you should remove these in favor of the types exported by this library.
```

Comment:
I'd put in a mention that these are optional to reassure people who don't want TS